inno_setup/gnucash-mingw64.iss

```diff
@@ -423,7 +423,7 @@ begin
 
   FileName := appdir + '\bin\guile.cmd' ;
   FileString := 'set GUILE_LOAD_PATH=' + pkgdatadir + '\scm;' + appdir + '\share\guile\2.2;%GUILE_LOAD_PATH%'#13#10 ;
-  FileString := FileString + 'start guile.exe %*'#13#10 ;
+  FileString := FileString + 'start "" "{app}\bin\guile.exe" %*'#13#10 ;
```

Comment:
Good catch. It occurs to me that this might have been behind the bugs that we used to get with users who'd also installed Lilypond, a music app that also uses Guile.

Reply:
Understand. Broader idea: Maybe `PATH` should be prepended with the GC bin folder first thing when firing up GC in order to force using executables shipped with GC, regardless of the calling environement?

Reply:
I got to wondering about why we even have guile.cmd. I'm pretty sure the only thing we shell out for nowadays is to run perl for Finance::Quote, so I went looking. Here's what I found:
```
commit 53c6782cf79a6c555f6c87065d77768f922dda6f
Author: Andreas Köhler <andi5.py@gmx.net>
Date:   Sat Dec 23 18:50:20 2006 +0000

    Use `cp -a' in dist.sh. Let inno create guile.bat (which sets
    GUILE_LOAD_PATH) and slibcat, because the user running GnuCash later on
    may not have the rights to do that.


    git-svn-id: svn+ssh://svn.gnucash.org/repo/gnucash/trunk@15251 57a11ea4-9604
-0410-9ed3-97b8803252fd
```

We set the paths in `gnucash\etc\gnucash\environment` now. Not only that but I just checked the installation directory and it isn't even installed. Let's just get rid of it.

So to $PATH: It would be harmless but I don't think we actually run anything from there, although we sure install a ton of them. Probably only the aqbanking-related ones are useful. This isn't the right repo anyway, it would be set in the `environment` file.

Reply:
> We set the paths in `gnucash\etc\gnucash\environment` now.

Just for those interested in this:  `environment` is generated cmake from [environment.in](https://github.com/Gnucash/gnucash/blob/maint/gnucash/environment.in) by cmake at configure time and at install time processed  by https://github.com/Gnucash/gnucash-on-windows/blob/1335548e0457dc4ebcda247eb2865431a76d7c44/inno_setup/gnucash-mingw64.iss#L443

> Not only that but I just checked the installation directory and it isn't even installed. Let's just get rid of it.

Yes, I cannot find `guile.cmd` in my installation folder, either.  I'll drop everything about it in the next push. (Umm, weird that inno refuses to create it albeit the instruction.)      
 
> So to $PATH: It would be harmless but I don't think we actually run anything from there, although we sure install a ton of them. Probably only the aqbanking-related ones are useful. This isn't the right repo anyway, it would be set in the `environment` file.

Understand. Because of [`PATH={GNC_BIN};{PATH}`](https://github.com/Gnucash/gnucash/blob/b5fdcfcb5b31a639d82cc79920ecf830c33ec450/gnucash/environment.in#L48-49)  `GNC_BIN` should be prepended to `PATH`.  If it works like that, then my theory about the crash in bug 797547  goes down. :(     

BTW: ... just seen that: Is the double slash in 
[`GUILE_LOAD_PATH={GNC_DATA}/scm;{GNC_HOME}/share/guile//@GUILE_EFFECTIVE_VERSION@;{GUILE_LIBS};{GUILE_LOAD_PATH}`](https://github.com/Gnucash/gnucash/blob/b5fdcfcb5b31a639d82cc79920ecf830c33ec450/gnucash/environment.in#L60) correct? My file explorer cannot digest any of them in paths.

Reply:
I don't remember offhand, but IIRC Windows Explorer doesn't like forward slashes at all. The CMD shell and Powershell take both forward and backward and don't usually care about doubled ones.